views/js/runner/plugins/tools/textToSpeech/textToSpeech.js

```diff
@@ -239,12 +239,28 @@ define([
              * Click to pronounce
              */
             clickToSpeak: function clickToSpeak() {
+                var self = this;
+
                 options.enableClickToSpeak = !options.enableClickToSpeak;
 
                 this._exec('enableClickToSpeak', options.enableClickToSpeak);
                 this._exec('enableContinuousReading', !options.enableClickToSpeak); // continuous reading is off when click to speak is on
                 this.trigger('clickToSpeak');
 
+                //adding each item a special class by presence of which normal click handling could be prevented and passed to click-to-speak handling
+                $('.qti-item').each(function() {
+                    var $_self = $(this);
+                    $_self.toggleClass('prevent-click-handler');
+                });
+
+                //we should disable click-to-speak while navigating through test, if it was enabled on some item page
+                //just to ensure that after other item load click-to-speak will function normally, and tao click handlers won't work on item-part click
+                $('.navi-box').on('click', 'a', function() {
+                    if ($('.qti-item.prevent-click-handler').length > 0) {
```

Comment:
The TestRunner provider exposes an areaBroker object that gives access to the DOM. Please use it.

```javascript
testRunner.getAreaBroker().getContentArea()
```

Reply:
So, actually, as I see this would be changes to smth like:
```
options['$navigationArea'].on('click', 'a', function() {
                    if (options['$contentArea'].find('.qti-item.prevent-click-handler').length > 0) {
```
Or am I wrong?

Reply:
Well, you should take at look at the build of the areaBroker to know more about the available areas and what are the actual targets:
https://github.com/oat-sa/extension-tao-testqti/blob/master/views/js/runner/provider/qti.js#L69

Reply:
Yes, as the TTS compoent does not have direct access to the TR, you must rely on delegated properties like that.

Reply:
Tank you for making it clearer for me :)